src/createResolveElements.js

```diff
@@ -61,13 +68,29 @@ export default function createResolveElements(environment) {
         params = route.prepareParams(params, routeMatch);
       }
 
+      return params;
+    });
+
+    const queryConfigs = routeMatches.map((routeMatch, i) => {
       const queries = matchQueries[i];
       if (!queries) {
         return null;
       }
 
-      const name = `__route_${i}_${routeIndices[i]}`;
-      return { name, queries, params };
+      return {
+        name: `__route_${i}_${routeIndices[i]}`,
+        queries,
+        params: routeParams[i],
```

Comment:
uh, why not mapping over `routeParams` instead of `routeMatches` so that you can just pass `params`